### ISSUE TITLE
Inline arrays of enums seems to break c-header-writing

### ISSUE THREAD

**Issue by lundmark** [wc-duck/datalibrary#137]
`diff -r 957609419922 datalibrary/src/dl_typelib_write_c_header.cpp
--- a/datalibrary/src/dl_typelib_write_c_header.cpp	Wed Jan 27 17:35:35 2021 +0100
+++ b/datalibrary/src/dl_typelib_write_c_header.cpp	Fri Jan 29 15:43:46 2021 +0100
@@ -610,6 +610,14 @@
 		case DL_TYPE_ATOM_INLINE_ARRAY:
 		{
 			dl_binary_writer_write_string_fmt(writer, "    ");
+
+			// If a member is marked as a struct it could also have been an enum that we didn't know about parse-time, patch it in that case.
+			if( member->storage == DL_TYPE_STORAGE_STRUCT )
+			{
+				if( const dl_enum_desc* edesc = dl_internal_find_enum( ctx, member->type_id ) )
+					member->storage = edesc->storage;
+			}
+
 			dl_context_write_type(ctx, member->storage, member->type_id, writer);
 			dl_binary_writer_write_string_fmt( writer, " %s[%u];\n", member->name, member->array_count );
 		}
`

This patch fixes an issue with writing a c-header with inline-arrays of enums. I expect there to be a better solution for this issue maybe generally, but unfortunately I did not have time to dive deeper for a better fix.

**Comment by wc-duck** [comment]
That solution do not really look ideal no... But I have some things to test now at least... I'll try to find some time to look at it during the week!

**Comment by lundmark** [comment]
Yeah the code was actually copy-paste from a different place. I think it's because DL doesn't require enums to be defined before used in the typelibrary.

**Comment by wc-duck** [comment]
Something like that yes... Do you have a typelib to repro the issue?

**Comment by lundmark** [comment]
Unfortunately I haven't been able to extract the parts of the typelib that causes this issue, and the typelibs are too huge and with too much secret-stuff so I can't send them either :/ But it *should* be easy to reproduce?

**Comment by wc-duck** [comment]
I can't seem to reproduce this... things I have tried is just inline arrays of enums (these tests has been there for a long while) and declaring the enum after the struct using it, that also works.
Could it be if the enum is from another typelib all together?

**Comment by lundmark** [comment]
Hi, sorry for the late answer.

Yes it is possible but unlikely that it's from a different typelib. Maybe it's possible to track down the other fix that I copied in order to get this working, and see what happens if you remove that one? The patch is basically a copy-paste from a different part of the code.
